### DISCUSSION TITLE
[Feature] Edit with External App

### DISCUSSION THREAD

**Discussion by kevincox** [immich-app/immich#13415] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

Right now the Immich Android only has a trivial photo editor (you can only crop) and no video editor. While a few in-app editing features are nice it would be great to have a button to open the asset in an external editor. Android has intents for this so any editor installed on the device can be used. This offers powerful flexibility as the user can use as simple and easy to use or as complex a photo editing app as they prefer.

A good example of this is [Aves](https://github.com/deckerst/aves):

![Screenshot_20241013-151331_Aves](https://github.com/user-attachments/assets/0094668c-6a4b-4b2d-95ab-d62f3bf498ea)![Screenshot_20241013-151339_Android System](https://github.com/user-attachments/assets/de29a9d7-ba0e-4ab9-9b11-f0025d58a4a4)

### Platform

- [ ] Server
- [x] Web
- [X] Mobile

**Comment by HSF-2003** [comment]
definitely need this, its the only thing keeping me from completing switching away from google photos

**Comment by ivanosh** [comment]
The only feature missing for me to remove Google Photos from my phone

**Comment by josx2** [comment]
Voting for this a well

**Comment by l-mb** [comment]
This would indeed be very beneficial. Editing the images separately also works (since the background sync will catch that), but it's not as nicely integrated as other solutions and is the only workflow that feels "off" compared to Google's builtin.

**Reply by ndesgranges** [reply]
Could you elaborate on the "since the background sync will catch that" ?

I tried to randomly modified some files for testing propose, and my modified versions always stay locally, Immich won't reupload it

**Reply by NirgaliDietrich** [reply]
Any chance your modified images are saved to a separate folder that isn't being backed up?

**Comment by WACOMalt** [comment]
This is a vital feature for me on Android. Unfortunately I didn't realize it was not implemented yet until I'd migrated my entire Google photos over.

I'm going to give a shot at implementing this myself this weekend. No promises, it's been at least 6 years since I've messed with Android development.. but I'll try. It seems like a simple addition. At least for pictures that are already synced locally on the phone.

**Reply by sjauijn** [reply]
well.....

> This is a vital feature for me on Android. Unfortunately I didn't realize it was not implemented yet until I'd migrated my entire Google photos over.
> 
> I'm going to give a shot at implementing this myself this weekend. No promises, it's been at least 6 years since I've messed with Android development.. but I'll try. It seems like a simple addition. At least for pictures that are already synced locally on the phone.

**Comment by freakyjoe8** [comment]
I think this could maybe a quick win till implementing all relevant editing features natively in the app is done. Also looking for such a possibility atm.

**Comment by andreaemonti** [comment]
I wanted to add that currently (app version 2.5.6 on android) editing a photo that has been synced on immich, using another gallery app (e.g. Samsung stock gallery) has the effect of having the image duplicated in immich: the original copy being only remote and the edited copy being available locally.

Independently of the "edit externally" button it would be nice to have the external edits being recognized. Or does this break anything else?

**Comment by snowfiend131** [comment]
Adding a +1 here.  A button to directly open an image in another photo app for editing would be a great help for workflow.  

The example of the Aves app functionality in the initial post is perfect - I recommend copying that feature implementation as closely as possible.

**Comment by anton-kozyr** [comment]
We  need more votes here 😅
